railties/lib/rails/commands/console/console_command.rb

```diff
@@ -12,6 +12,47 @@ def filter_backtrace(bt)
       end
     end
 
+    class IRBConsole
+      def initialize
+        require "irb"
+        require "irb/completion"
+        IRB::WorkSpace.prepend(BacktraceCleaner)
+
+        if !Rails.env.local?
+          # Use env var here so users can override them with env var too
+          ENV["IRB_USE_AUTOCOMPLETE"] ||= "false"
+        end
+
+        IRB
+      end
+
+      def name
+        "IRB"
+      end
+
+      def start
+        IRB.setup(nil)
+
+        env =
+          if Rails.env == "development"
+            "dev"
+          else
+            Rails.env.first(4)
+          end
```

Comment:
`Rails.env.first(4)` might be a bit too generic.  For example, `staging` would be truncated as `stag`, which may not be clear.  Also, if the user has similarly named environments (e.g. `staging_alpha` and `staging_beta`), there could be a collision.  Though I guess it's debatable whether that is preferable to a static prompt.

Should we start with just special casing the `development`, `test`, and `production` environments?

Reply:
Since `Rails.env` is a `EnvironmentInquirer`, I think we could have `Rails.env.short_name` or something, and have that logic there. 

I think it make sense to have short name for known values and just return to complete thing for unknown values.

Reply:
I decided to adopt @byroot's suggestion here. But since we'll colorise the env too, I feel a case statement here is actually simpler than having `Rails.env.short_name`.